scripts/modules/MsIndustryLinks/templates/ingest/IngestionWorkflow.psm1

```diff
@@ -92,6 +94,120 @@ function New-LogicAppIngestionWorkflow {
+    if ($dataSink -eq "dataverse") {
+        $template = New-FlowDataverseIngestionWorkflow -DataSinkConfig $DataSinkConfig
+    }
+    elseif ($dataSink -eq "customconnector") {
+        $template = New-FlowCustomConnectorIngestionWorkflow -DataSinkConfig $DataSinkConfig -AuthConfigFile $AuthConfigFile
+    }
+    else {
+        throw "The data sink type, $($DataSinkConfig.type), is not supported. Please choose from: CustomConnector, Dataverse."
+    }
+
+    $template.name = [guid]::NewGuid().ToString()
+    $template.properties.displayName = $DataSinkConfig.name
+
+    return $template
+}
+
+function New-FlowCustomConnectorIngestionWorkflow {
+    param (
+        [Parameter(Mandatory = $true, HelpMessage = "The data sink workflow configuration object.")]
+        [object] $DataSinkConfig,
+        [Parameter(Mandatory = $false, HelpMessage = "The path to the authentication configuration JSON file.")]
+        [string] $AuthConfigFile
+    )
+
+    $baseTemplate = Get-Content $PSScriptRoot/templates/flow_base.json | ConvertFrom-Json
+    $definition = Get-Content $PSScriptRoot/templates/ingest/customconnector/flow_customconnector.json | ConvertFrom-Json
+
+    # Set variables based on whether the custom connector has been certified
+    if ($DataSinkConfig.isCertified) {
+        $apiId = "/providers/Microsoft.PowerApps/apis/$($DataSinkConfig.connection.apiName)"
+
+        $apiConfig = @{
+            name = $DataSinkConfig.connection.apiName
+        }
+
+        $connectionReferenceName = "$($DataSinkConfig.connection.apiName)_ref"
+    }
+    else {
+        if ($null -eq $AuthConfigFile -or $AuthConfigFile -eq "") {
+            throw "The authentication configuration file (-AuthConfigFile) is required for uncertified custom connectors."
+        }
+
+        $apiId = ""
+
+        $authConfig = Get-Content $AuthConfigFile | ConvertFrom-Json
+        $connectorIdentifiers = Get-ConnectorIdentifiers -ConnectorId $DataSinkConfig.connection.connectorId -AuthConfig $authConfig
+
+        $apiConfig = $connectorIdentifiers
+        $connectionReferenceName = "$($connectorIdentifiers.logicalName)_ref"
+    }
+
+    $inputType = $DataSinkConfig.inputType.ToLower()
+    # Configure the flow based on whether the custom connector accepts a single record or an array of records
+    if ($inputType -eq "array") {
+        # Parameters are required to pass the data into the custom connector
+        $parameters = $DataSinkConfig.parameters
+        if ($null -eq $parameters -or 0 -eq @($parameters.psobject.Properties).Count) {
+            throw "No parameters were specified for the custom connector data sink."
+        }
+
+        # Remove the functionality for a custom connector to ingest a single record
+        $definition.actions.PSObject.Properties.Remove("For_each_item")
+
+        # Map the input data to the format of the custom connector API
+        $definition.actions.Map_data.inputs.select = $DataSinkConfig.mapping
```

Comment:
Why is this needed for array but not object?

Reply:
Is there another more efficient way to remap an array in flows that I have missed?

Reply:
Sorry, I saw "$DataSinkConfig.mapping" only for the array version. Guess I missed it in the object version.